changelog/7616.improvement.md

```diff
@@ -0,0 +1,8 @@
+Added cross-entropy loss over sigmoid of all similarity values to constrain them in an approximate range in `DotProductLoss`.
+
+This affects the default behaviour of the loss function(`loss_type=cross_entropy`) inside machine learning (ML) components - `DIETClassifier`, `ResponseSelector` and `TEDPolicy`.
```

Comment:
Please also mention in the migration guide that fallback thresholds might be affected by this change (regardless of the option really, because we add sigmoids by default)